cuqi/problem/_problem.py

```diff
@@ -588,7 +585,15 @@ def gradfunc(x): return -density.gradient(x)
             if disp: print("Optimizing with approximate gradients.") 
 
         # Compute point estimate
-        solver = cuqi.solver.L_BFGS_B(func, x0, gradfunc=gradfunc)
+        if self._check_posterior(self, Cauchy_diff, must_have_gradient=True): # Use L-BFGS-B for Cauchy_diff prior as it has better performance for this multi-modal posterior
+            if disp: print(f"Using scipy.optimize.L_BFGS_B on negative log of {density.__class__.__name__}")
```

Comment:
Something we might want to consider in general in the future: I suggest adding an issue to use logging library for information/error messages etc. The user can then specify global setting on the level of information displayed.
e.g. https://towardsdatascience.com/stop-using-print-and-start-using-logging-a3f50bc8ab0